service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/cmo/DraftOrdersApprovedEventHandler.java

```diff
@@ -143,32 +185,55 @@ public void sendNotificationToCafcassViaSendGrid(final DraftOrdersApproved event
         CaseData caseData = event.getCaseData();
 
         if (CafcassHelper.isNotifyingCafcassEngland(caseData, cafcassLookupConfiguration)) {
+            List<HearingOrder> approvedOrders = new ArrayList<>();
+            approvedOrders.addAll(event.getApprovedOrders());
+            approvedOrders.addAll(getConfidentialOrderToBeSentToCafcass(event));
+
+            if (approvedOrders.isEmpty()) {
+                log.info("Only confidential orders uploaded and not accessible by Cafcass. "
+                         + "Skip notifying Cafcass England");
+                return;
+            }
+
             LocalDateTime hearingStartDate = findElement(caseData.getLastHearingOrderDraftsHearingId(),
                     caseData.getHearingDetails())
                     .map(Element::getValue)
                     .map(HearingBooking::getStartDate)
                     .orElse(null);
 
-            event.getApprovedOrders()
-                .forEach(hearingOrder ->
+            approvedOrders.forEach(hearingOrder ->
                     cafcassNotificationService.sendEmail(caseData,
-                            of(hearingOrder.getOrder()),
+                            Set.of(hearingOrder.isConfidentialOrder() ? hearingOrder.getOrderConfidential()
+                                : hearingOrder.getOrder()),
                             ORDER,
                             OrderCafcassData.builder()
                                     .documentName(hearingOrder.getTitle())
                                     .hearingDate(hearingStartDate)
                                     .orderApprovalDate(hearingOrder.getDateIssued())
                                     .build()
-                        )
-                );
+                    ));
         }
     }
 
+    private List<HearingOrder> getConfidentialOrderToBeSentToCafcass(final DraftOrdersApproved event) {
+        CaseData caseData = event.getCaseData();
+        List<Element<GeneratedOrder>> allChildConfidentialOrders = caseData.getConfidentialOrders()
+            .getAllChildConfidentialOrders();
+        return event.getApprovedConfidentialOrders().stream()
+            .filter(order -> findElement(order.getId(), allChildConfidentialOrders).isPresent())
+            .map(Element::getValue)
+            .toList();
+    }
+
     @Async
     @EventListener
     public void sendNotificationToDigitalRepresentatives(final DraftOrdersApproved event) {
         CaseData caseData = event.getCaseData();
         List<HearingOrder> approvedOrders = event.getApprovedOrders();
+        if (event.getApprovedOrders().isEmpty()) {
+            log.info("No non-confidential approved orders. skip sendDocumentToPostRecipients");
```

Comment:
Change to `sendNotificationToDigitalRepresentatives`

Reply:
nice spotting, updated